src/main/java/org/gridsuite/study/server/dto/ShortCircuitCustomParameters.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2021, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package org.gridsuite.study.server.dto;
+
+import com.powsybl.shortcircuit.InitialVoltageProfileMode;
+import com.powsybl.shortcircuit.StudyType;
+import lombok.*;
+
+/**
+ * @author AJELLAL Ali <ali.ajellal@rte-france.com>
+ */
+
+@NoArgsConstructor
+@AllArgsConstructor
+@Getter
+@Setter
+@Builder
+public class ShortCircuitCustomParameters {
```

Comment:
Why did you create those CustomParameters and you do not use the parameters of powsybl ?

Reply:
This is for two reasons: 
1- Based on the UI I need to send the predefined setting selected by the user and I can't do that using  the parameters of powsybl(no field is  mapped to the predefined setting).
2- Using the parameters of powsybl class I will be forced to send((from the front to back) the **voltageRanges** each time the  **initialVoltageProfileMode** is equals to CONFIGURED otherwise it crashes during deserialization.

Check : 
- powsybl **ShortCircuitParametersDeserializer** class
[https://github.com/powsybl/powsybl-core/blob/ecb61c4be02c5f113cd0568e7f69bde205a017b8/shortcircuit-api/src/main/java/com/powsybl/shortcircuit/json/ShortCircuitParametersDeserializer.java](url)

and the methode **validate()** of the class **ShortCircuitParameters.java**